server/game/cards/08.5-TFM/TheSongOfTheSeven.js

```diff
@@ -0,0 +1,38 @@
+const DrawCard = require('../../drawcard.js');
+
+class TheSongOfTheSeven extends DrawCard {
+    setupCardAbilities(ability) {
+        this.reaction({
+            max: ability.limit.perRound(1),
+            when: {
+                onDominanceDetermined: event =>
+                    this.controller !== event.winner &&
+                    event.winner.faction.power >= 2 &&
+                    this.hasTheSeven()
```

Comment:
This is unneeded as the target API already makes sure the card won't trigger if there's no legal target.

Reply:
I guess eventually, if you keep repeating this to me, I'll learn about this, but no promises :-)
(all comments addressed)